File: client/src/types/room.ts
export type RoomType = string
export type RoomStatus = 'available' | 'occupied' | 'maintenance'

export interface ApiRoomType {
  id: string
  name: string
  sort_order: number
  created_at: string
  updated_at: string
}

export interface ApiRoom {
  id: string
  room_number: string
  floor: number
  type: RoomType
  status: RoomStatus
  rent_price: number
  description: string
  created_at: string
  updated_at: string
  updated_by: string
  updated_by_name: string
}

export interface CreateRoomPayload {
  room_number: string
  floor: number
  type: RoomType
  status: RoomStatus
  rent_price: number
  description: string
}

export interface UpdateRoomPayload {
  room_number?: string
  floor?: number
  type?: RoomType
  status?: RoomStatus
  rent_price?: number
  description?: string
}

File: backend/internal/features/room/domain/room.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type RoomStatus string

const (
	RoomStatusAvailable   RoomStatus = "available"
	RoomStatusOccupied    RoomStatus = "occupied"
	RoomStatusMaintenance RoomStatus = "maintenance"
)

func (s RoomStatus) Valid() bool {
	switch s {
	case RoomStatusAvailable, RoomStatusOccupied, RoomStatusMaintenance:
		return true
	}
	return false
}

type Room struct {
	ID            uuid.UUID  `json:"id"`
	DormitoryID   uuid.UUID  `json:"dormitory_id"`
	DormitoryName string     `json:"dormitory_name,omitempty"`
	RoomTypeID    uuid.UUID  `json:"room_type_id"`
	RoomTypeName  string     `json:"room_type_name,omitempty"`
	RoomTypePrice float64    `json:"room_type_price"`
	RoomNumber    string     `json:"room_number"`
	Floor         int        `json:"floor"`
	Status        RoomStatus `json:"status"`
	IsActive      bool       `json:"is_active"`
	CreatedBy     *uuid.UUID `json:"created_by,omitempty"`
	UpdatedBy     *uuid.UUID `json:"updated_by,omitempty"`
	CreatedAt     time.Time  `json:"created_at"`
	UpdatedAt     time.Time  `json:"updated_at"`
}
